### PR TITLE
Make the golint plugin skip generated files.

### PR DESCRIPTION
**Pull request by spxtr**
@nlandolfi 
/area prow
/assign @rmmh

### CODE REVIEW

File: prow/plugins/golint/golint.go [hunk 1]
```diff
@@ -56,18 +59,28 @@ func handleIC(pc plugins.PluginClient, ic github.IssueCommentEvent) error {
...
-// that are modified in the PR excluding vendor/.
-func modifiedGoFiles(ghc githubClient, org, repo string, number int) (map[string]string, error) {
+// that are modified in the PR excluding vendor/ and generated files.
+func modifiedGoFiles(ghc githubClient, org, repo string, number int, sha string) (map[string]string, error) {
 	changes, err := ghc.GetPullRequestChanges(org, repo, number)
 	if err != nil {
 		return nil, err
 	}
 
+	gfg, err := genfiles.NewGenFilesGroup(ghc, org, repo, sha)
```
Comment (nlandolfi) [inline prow/plugins/golint/golint.go]: would be nice if in the future we changed this to something like `genfiles.NewGroup`

File: prow/plugins/golint/golint_test.go [hunk 2]
```diff
@@ -157,6 +178,7 @@ func TestLint(t *testing.T) {
 	if len(gh.comment.Comments) != maxComments {
 		t.Fatalf("Expected %d comments, got %d: %v", maxComments, len(gh.comment.Comments), gh.comment.Comments)
 	}
+
```
Comment (nlandolfi) [inline prow/plugins/golint/golint_test.go]: nit: could possibly delete new line

**Comment by rmmh** [comment]
I'm slightly concerned about how many API calls this will take-- the current mungegithub caches generated_files.

/lgtm

File: prow/plugins/golint/golint.go (hunk 1 above)
Comment (spxtr) [reply prow/plugins/golint/golint.go]: Done.

File: prow/plugins/golint/golint_test.go (hunk 2 above)
Comment (spxtr) [reply prow/plugins/golint/golint_test.go]: Done.

**Comment by spxtr** [comment]
> I'm slightly concerned about how many API calls this will take-- the current mungegithub caches generated_files.

We can also cache generated_files without too much difficulty if necessary. I don't think the one extra token will be a problem.

**Pull request merged**